app/stac_api/models.py

```diff
@@ -197,27 +190,469 @@ def update_geoadmin_variants(self, asset_geoadmin_variant, asset_proj_epsg, asse
+        (in this case None, as it is a new item)
+        :param new_start_datetime: item's updated value for properties_start_datetime
+        :param old_end_datetime: item's old value for properties_end_datetime
+        (in this case None, as it is a new item)
+        :param new_end_datetime: item's updated value for properties_end_datetime
+        :param item_id: the id of the item being treated
         '''
-
+        logger.debug('Inserting item %s in collection %s', item_id, self.collection_name)
         if self.cache_start_datetime is None:
-            self.cache_start_datetime = item_properties_start_datetime
-            self.save()
-        elif item_properties_start_datetime < self.cache_start_datetime:
-            self.cache_start_datetime = item_properties_start_datetime
-            self.save()
-        elif self.cache_end_datetime is None:
-            self.cache_end_datetime = item_properties_end_datetime
-            self.save()
-        elif item_properties_end_datetime > self.cache_end_datetime:
-            self.cache_end_datetime = item_properties_end_datetime
-            self.save()
+            # first item in collection, as cache_start_datetime is None:
+            self.cache_start_datetime = new_start_datetime
+        elif self.cache_start_datetime > new_start_datetime:
+            # new item starts earlier that current collection range starts
+            self.cache_start_datetime = new_start_datetime
+
+        if self.cache_end_datetime is None:
+            #first item in collection, as cache_start_datetime is None
+            self.cache_end_datetime = new_end_datetime
+        elif self.cache_end_datetime < new_end_datetime:
+            # new item starts after current collection's range ends
+            self.cache_end_datetime = new_end_datetime
+
+        self.save()
+
+    def update_item_start_datetime(
+        self,
+        old_start_datetime,
+        new_start_datetime,
+        old_end_datetime,
+        new_end_datetime,
+        item_id,
+        qs=None
+    ):
+        '''
+        This function is called from within update_temporal_extent() when an
+        item in the collection is updated to check if the collection's
+        start_datetime needs to be updated.
+        :param old_start_datetime: item's old value for properties_start_datetime
+        :param new_start_datetime: item's updated value for properties_start_datetime
+        :param old_end_datetime: item's old value for properties_end_datetime
+        :param new_end_datetime: item's updated value for properties_end_datetime
+        :param item_id: the id of the item being treated
+        :param qs: (optional) queryset with all items of the collection excluding the one
+        being updated.
+        '''
+        logger.debug(
+            'Updating item %s in collection (adjusting start_datetime) %s',
+            item_id,
+            self.collection_name
+        )
+        if old_start_datetime != new_start_datetime:
```

Comment:
Noticed that this can be removed, as it is redundant. This function is only called, when this condition is true (see line 546)

Reply:
removed redundant if statement